### PR TITLE
WorkflowBackNavigation: Add isBackNavigation param into Signalr model

### PR DESCRIPTION
**Pull request by reyhanlioglu**
## Summary by CodeRabbit

- **New Features**
	- Enhanced navigation data model to include back navigation detection.

- **Refactor**
	- Updated networking classes to support new navigation feature.

- **Documentation**
	- Added serialization annotations for new data model fields.

### CODE REVIEW

**Review by yildirayyorulmaz — APPROVED** [review]

**Review by ayhncvdr2 — APPROVED** [review]

File: lib/neo_core.dart [hunk 1]
```diff
@@ -15,7 +15,6 @@ import 'package:neo_core/core/storage/neo_core_secure_storage.dart';
 export 'core/bus/neo_bus.dart';
 export 'core/network/neo_network.dart';
 export 'core/storage/neo_storage.dart';
-export 'core/util/neo_util.dart';
```
Comment (yildirayyorulmaz) [inline lib/neo_core.dart:18]: export 'device_util.dart';
eklenebilir

File: lib/core/storage/neo_core_secure_storage.dart [hunk 2]
```diff
@@ -12,7 +12,7 @@
 
 import 'package:flutter_secure_storage/flutter_secure_storage.dart';
 import 'package:neo_core/core/encryption/jwt_decoder.dart';
-import 'package:neo_core/neo_core.dart';
+import 'package:neo_core/core/util/device_util.dart';
```
Comment (yildirayyorulmaz) [inline lib/core/storage/neo_core_secure_storage.dart:15]: import 'package:neo_core/neo_core.dart';
kalsin bence

device_util.dart'i neo_core'a ekleyebilriiz

File: lib/core/network/managers/neo_network_manager.dart [hunk 3]
```diff
@@ -19,6 +19,7 @@ import 'package:neo_core/core/network/models/http_auth_response.dart';
 import 'package:neo_core/core/network/models/http_method.dart';
 import 'package:neo_core/core/network/models/neo_http_call.dart';
 import 'package:neo_core/core/network/models/neo_network_header_key.dart';
+import 'package:neo_core/core/util/device_util.dart';
```
Comment (yildirayyorulmaz) [inline lib/core/network/managers/neo_network_manager.dart:22]: ayni sekilde buna da gerek yok

**Review by mcanyigincommencis — APPROVED** [review]

File: lib/core/storage/neo_core_secure_storage.dart (hunk 2 above)
Comment (reyhanlioglu) [reply lib/core/storage/neo_core_secure_storage.dart]: Done

File: lib/core/network/managers/neo_network_manager.dart (hunk 3 above)
Comment (reyhanlioglu) [reply lib/core/network/managers/neo_network_manager.dart]: Done

File: lib/neo_core.dart (hunk 1 above)
Comment (reyhanlioglu) [reply lib/neo_core.dart:18]: Done

**Pull request merged**